certmanager/generate_cert.go

```diff
@@ -174,18 +189,40 @@ func genCertTemplate(options CertOptions) x509.Certificate {
 		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
 		ExtKeyUsage:           []x509.ExtKeyUsage{extKeyUsage},
 		BasicConstraintsValid: true,
-	}
-	hosts := strings.Split(options.Host, ",")
-	for _, h := range hosts {
-		if ip := net.ParseIP(h); ip != nil {
-			template.IPAddresses = append(template.IPAddresses, ip)
-		} else {
-			template.DNSNames = append(template.DNSNames, h)
-		}
+		ExtraExtensions:       []pkix.Extension{sanExt},
 	}
 	if options.IsCA {
 		template.IsCA = true
 		template.KeyUsage |= x509.KeyUsageCertSign
 	}
 	return template
 }
+
+func buildSubjectAltNameExtension(host string) pkix.Extension {
```

Comment:
also update https://github.com/istio/auth/tree/master/cmd/generate_cert to use URI?